MatchAndPlan/src/transferCenterRoutePlanning/Util/LatLng.java

```diff
@@ -0,0 +1,35 @@
+package transferCenterRoutePlanning.Util;
+
+public  class LatLng {
+
+    public final double latitude;
+    public final double longitude;
+    public final double latitudeE6;
+    public final double longitudeE6;
+
+
+    public LatLng(double var1, double var3) {
+        if (!Double.isNaN(var1) && !Double.isNaN(var3) && !Double.isInfinite(var1) && !Double.isInfinite(var3)) {
+            double var5 = var1 * 1000000.0D;
+            double var7 = var3 * 1000000.0D;
+            this.latitudeE6 = var5;
+            this.longitudeE6 = var7;
+            this.latitude = var5 / 1000000.0D;
+            this.longitude = var7 / 1000000.0D;
+        } else {
+            this.latitudeE6 = 0.0D;
+            this.longitudeE6 = 0.0D;
+            this.latitude = 0.0D;
+            this.longitude = 0.0D;
+        }
+    }
+
+
+    public String toString() {
+        String var1 = new String("latitude: ");
```

Comment:
I detect that this code is problematic. According to the [Performance (PERFORMANCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#performance-performance), [Dm: Method invokes inefficient new String(String) constructor (DM_STRING_CTOR)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dm-method-invokes-inefficient-new-string-string-constructor-dm-string-ctor).
Using the java.lang.String(String) constructor wastes memory because the object so constructed will be functionally indistinguishable from the String passed as a parameter.  Just use the argument String directly.